migration/usage-metric-migration/src/main/java/io/camunda/migration/usagemetric/TUMetricMigrator.java

```diff
@@ -0,0 +1,259 @@
+              tasklistTaskImportPositionQuery(),
+              ImportPositionEntity.class);
+      final var finished =
+          !importPositions.isEmpty()
+              && importPositions.stream().allMatch(ImportPositionEntity::getCompleted);
+      if (finished) {
+        LOG.info("Tasklist importers have concluded.");
+        return true;
+      } else if (Instant.now().isAfter(timeout)) {
+        LOG.error(
+            "Importer did not finish within the timeout of {}. Aborting migration.",
+            configuration.getTimeout());
+        throw new MigrationTimeoutException(
+            "Importer did not finish within the timeout of " + configuration.getTimeout(), false);
+      }
+    } catch (final MigrationException e) {
+      LOG.error("Failed to check whether importers are done", e);
+    }
+    return false;
+  }
+
+  private String reindex() {
+    final var minEventTime = OffsetDateTime.now().minusYears(2);
+    try {
+      final var assignees =
+          client.getAllAssigneesInMetrics(
+              indexDescriptors.get(TasklistMetricIndex.class).getFullQualifiedName());
+      final Map<String, Long> hashedAssignees = hashAssignees(assignees);
+      return client.reindex(
+          indexDescriptors.get(TasklistMetricIndex.class).getFullQualifiedName(),
+          indexDescriptors.get(UsageMetricTUIndex.class).getFullQualifiedName(),
+          eventDateQuery(minEventTime),
+          SCRIPT,
+          Map.of(PARAMS_KEY, hashedAssignees));
+    } catch (final IOException e) {
+      throw new MigrationException("Failed to start reindexing task", e);
+    }
+  }
+
+  private void monitorReindexTask(final String taskId) throws Exception {
+    final var retryDecorator = busyRetryDecorator();
+    metricRegistry.measureTasklistReindexTask(
+        () ->
+            retryDecorator.decorate(
+                "Wait for reindex to be completed", () -> pollReindexTask(taskId), done -> !done));
+  }
+
+  private boolean pollReindexTask(final String taskId) {
+    try {
+      final var status = client.getTask(taskId);
+      if (status.completed()) {
+        LOG.info("Reindex task {} completed successfully", taskId);
+        persistStatus(status.taskId(), true);
+        return true;
+      }
+    } catch (final Exception e) {
+      LOG.error("Failed to acquire status of reindexing task", e);
+    }
+    return false;
+  }
+
+  private Map<String, Long> hashAssignees(final Collection<String> assignees) {
+    return assignees.stream().collect(Collectors.toMap(assignee -> assignee, this::hashAssignee));
+  }
+
+  private long hashAssignee(final String assignee) {
+    return Hashing.murmur3_128().hashString(assignee, StandardCharsets.UTF_8).asLong();
+  }
+
+  public SearchQuery eventDateQuery(final OffsetDateTime minEventTime) {
+    return and(
+        SearchQueryBuilders.dateTimeOperations(
+            MetricIndex.EVENT_TIME, List.of(Operation.gt(minEventTime))));
+  }
```

Comment:
❓ Should we add a filter the `event="task_completed_by_assignee"`? I don't think there will any other event but just in case.